test/authorization/service/role.service.test.ts

```diff
@@ -78,170 +95,226 @@ describe('test Role Service', () => {
         { provide: 'RedisCacheService', useValue: redisCacheService },
         { provide: 'SearchService', useValue: searchService },
         {
-          provide: Connection,
-          useValue: connectionMock,
+          provide: DataSource,
+          useValue: mockDataSource,
         },
       ],
     }).compile();
-    roleService = moduleRef.get<RoleService>(RoleService);
-  });
 
-  it('should get all roles', async () => {
-    rolesRepository.createQueryBuilder().returns(rolesQueryBuilder);
-    rolesQueryBuilder.getManyAndCount().returns(Promise.resolve([roles, 1]));
-    const resp = await roleService.getAllRoles();
-    expect(resp).toEqual([roles, 1]);
+    roleService = moduleRef.get(RoleService);
+    roleRepository = moduleRef.get(RoleRepository);
+    permissionRepository = moduleRef.get(PermissionRepository);
+
+    createQueryBuilderMock = roleRepository.createQueryBuilder = jest
+      .fn()
+      .mockReturnValue({
+        leftJoinAndSelect: jest.fn().mockReturnThis(),
+        where: jest.fn().mockReturnThis(),
+        getManyAndCount: (getManyAndCountMock = jest.fn()),
+      });
+
+    findOneByMock = roleRepository.findOneBy = jest.fn();
+    saveMock = roleRepository.save = jest.fn();
+    updateMock = roleRepository.update = jest.fn();
   });
 
-  it('should get a role by id', async () => {
-    rolesRepository
-      .findOneBy({ id: 'ae032b1b-cc3c-4e44-9197-276ca877a7f8' })
-      .returns(Promise.resolve(roles[0]));
-    const resp = await roleService.getRoleById(
-      'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-    );
-    expect(resp).toEqual(roles[0]);
+  describe('getAllRoles', () => {
+    it('should get all roles when no parameters are passed', async () => {
+      getManyAndCountMock.mockReturnValue([roles, 1]);
+
+      const result = await roleService.getAllRoles();
+
+      expect(createQueryBuilderMock.mock.calls[0][0]).toStrictEqual('role');
+
+      expect(createQueryBuilderMock().where).not.toBeCalled();
+
+      expect(getManyAndCountMock).toBeCalledTimes(1);
+
+      expect(result).toEqual([roles, 1]);
+    });
   });
 
-  it('should create a role', async () => {
-    const input: NewRoleInput = {
-      name: 'Test1',
-    };
-    rolesRepository.create(input).returns(roles[0]);
+  describe('getRoleById', () => {
+    it('should get a role for valid id', async () => {
+      findOneByMock.mockResolvedValue(roles[0]);
+
+      const result = await roleService.getRoleById(VALID_ROLE_ID);
+
+      expect(findOneByMock).toBeCalledWith({
+        id: VALID_ROLE_ID,
+      });
+
+      expect(result).toEqual(roles[0]);
+    });
+
+    it('should throw exception for invalid id', async () => {
+      let result;
```

Comment:
Shall we remove this variable since it's not in use.

Reply:
Will fix service related tests in a separate PR.